hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/SqlLimitTest.java

```diff
@@ -0,0 +1,158 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql;
+
+import com.hazelcast.jet.sql.impl.connector.test.TestBatchSqlConnector;
+import com.hazelcast.sql.HazelcastSqlException;
+import com.hazelcast.sql.SqlService;
+import com.hazelcast.sql.SqlStatement;
+import com.hazelcast.sql.impl.SqlErrorCode;
+import com.hazelcast.sql.impl.type.QueryDataType;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.Collections;
+
+import static java.util.Arrays.asList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class SqlLimitTest extends SqlTestSupport {
+
+    private static SqlService sqlService;
+
+    @BeforeClass
+    public static void setUpClass() {
+        initialize(2, null);
+        sqlService = instance().getSql();
+    }
+
+    @Test
+    public void limitOverTable() {
+        String tableName = createTable(
+                new String[]{"Alice", "1"},
+                new String[]{"Bob", "2"},
+                new String[]{"Joey", "3"}
+        );
+
+        assertContainsOnlyOneOfRows(
+                "SELECT name FROM " + tableName + " LIMIT 1",
+                new Row("Alice"), new Row("Bob"), new Row("Joey")
+        );
+
+        assertContainsSubsetOfRows(
+                "SELECT name FROM " + tableName + " LIMIT 2",
+                2,
+                new Row("Alice"), new Row("Bob"), new Row("Joey")
+        );
+
+        assertRowsAnyOrder(
+                "SELECT name FROM " + tableName + " LIMIT 5",
+                asList(new Row("Alice"), new Row("Bob"), new Row("Joey"))
+        );
+    }
+
+    @Test
+    public void negativeLimitValue() {
+        String tableName = createTable(
+                new String[]{"Alice", "1"},
+                new String[]{"Bob", "2"},
+                new String[]{"Joey", "3"}
+        );
+
+        checkFailure0("SELECT name FROM " + tableName + " LIMIT -10", SqlErrorCode.PARSING, "Encountered \"-\"");
+    }
+
+    @Test
+    public void floatNumber_asLimitValue() {
+        String tableName = createTable(
+                new String[]{"Alice", "1"},
+                new String[]{"Bob", "2"},
+                new String[]{"Joey", "3"}
+        );
+
+        assertRowsAnyOrder(
+                "SELECT name FROM " + tableName + " LIMIT 5.2",
```

Comment:
Weird, we get parsing error for `-10`, but `5.2` works fine? In that case I'd like to see a test where the number actually limits, I guess `LIMIT 2.99` will return 2 rows out of 3.

Reply:
holy crap, this is Postgres example:
```
postgres=# select * from numbers ;
 col_si | col_i | col_bi
--------+-------+--------
    100 |   100 |    100
    100 |   100 |    100
      1 |     2 |      3
      1 |     2 |      3
      1 |     2 |      3
      1 |     2 |      3
      1 |     2 |      3
(7 rows)

postgres=# select * from numbers limit 2.99;
 col_si | col_i | col_bi
--------+-------+--------
    100 |   100 |    100
    100 |   100 |    100
      1 |     2 |      3
(3 rows)

postgres=# select * from numbers limit 2.5;
 col_si | col_i | col_bi
--------+-------+--------
    100 |   100 |    100
    100 |   100 |    100
      1 |     2 |      3
(3 rows)

postgres=# select * from numbers limit 2.45;
 col_si | col_i | col_bi
--------+-------+--------
    100 |   100 |    100
    100 |   100 |    100
(2 rows)
```